Source/CarthageKit/GitHub.swift

```diff
@@ -266,6 +284,14 @@ internal struct GitHubCredentials {
 	}
 }
 
+internal func loadGitHubCredentials(accessToken: GitHubAccessToken?) -> SignalProducer<GitHubCredentials?, CarthageError> {
```

Comment:
Would you be up for changing this method to something like this?

```swift
internal func loadGitHubAuthorization() -> SignalProducer<String?, CarthageError>
```

Then we could get rid of `GitHubCredentials`, `AccessTokenGitHubCredentials`, and `BasicGitHubCredentials`. All we really care about is the authorization header value. We don't need to pass around the actual credentials; it's just adding complexity.